lib/sisimai/bite/email/imailserver.rb

```diff
@@ -0,0 +1,181 @@
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            break if readcursor & Indicators[:'message-rfc822'] > 0
+
+            # Unknown user: kijitora@example.com
+            #
+            # Original message follows.
+            v = dscontents[-1]
+
+            if cv = e.match(/\A(.+)[ ](.+)[:][ \t]*([^ ]+[@][^ ]+)/)
+              # Unknown user: kijitora@example.com
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['diagnosis'] = cv[1] + ' ' + cv[2]
+              v['recipient'] = cv[3]
+              recipients += 1
+
+            elsif cv = e.match(/\Aundeliverable[ ]+to[ ]+(.+)\z/)
+              # undeliverable to kijitora@example.com
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            else
+              # Other error message text
+              v['alterrors']  += ' ' + e if v['alterrors']
+              if e =~ Re1[:error]
+                # Body of message generated response:
+                v['alterrors'] = e
+              end
+            end
+          end
+        end
+        return nil if recipients.zero?
+        require 'sisimai/string'
+
+        dscontents.map do |e|
+          e['agent'] = self.smtpagent
+
+          if e['alterrors'] && e['alterrors'].size > 0
+            # Copy alternative error message
+            if e['diagnosis']
+              e['diagnosis'] = e['alterrors'] + ' ' + e['diagnosis']
+            else
+              e['diagnosis'] = e['alterrors']
+            end
+            e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+            e.delete('alterrors')
+          end
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          ReSMTP.each_key do |r|
+            # Detect SMTP command from the message
+            next unless e['diagnosis'] =~ ReSMTP[r]
```

Comment:
__[rubocop]__
* Layout/IndentationWidth: Use 2 (not -15) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

<!-- Comment by [SideCI](https://sideci.com) -->